kubernetes/src/kubernetes/config/incluster_config.rb

```diff
@@ -62,19 +65,42 @@ def token_file
       @token_file
     end
 
+    def token_refresh_period
+      @token_refresh_period ||= TOKEN_REFRESH_PERIOD
+      @token_refresh_period
+    end
+
     def load_token
       File.open(token_file) do |io|
         self.token = io.read.chomp
+        self.token_expires_at = Time.now + token_refresh_period
       end
     end
 
-    def configure(configuration)
+    # rubocop:disable Metrics/AbcSize
+    def configure(configuration, refresh_token: true)
       validate
       load_token
       configuration.api_key['authorization'] = "Bearer #{token}"
       configuration.scheme = 'https'
       configuration.host = "#{host}:#{port}"
       configuration.ssl_ca_cert = ca_cert
+      return unless refresh_token
+
+      Configuration.instance_variable_set(:@in_cluster_config, self)
+      Configuration.prepend(Module.new do
+        # rubocop:disable Metrics/LineLength
+        def api_key_with_prefix(identifier)
+          in_cluster_config = self.class.instance_variable_get(:@in_cluster_config)
+          if identifier == 'authorization' && @api_key.key?(identifier) && in_cluster_config.token_expires_at <= Time.now
```

Comment:
The fact that the parameter is `refresh_token` but it's actually conditional is a little weird.

Perhaps name the parameter `try_refresh_token`?

Reply:
done